src/java/org/apache/cassandra/service/accord/async/AsyncWriter.java

```diff
@@ -253,7 +253,7 @@ public boolean save(AsyncContext context, BiConsumer<Object, Throwable> callback
 
                     state = State.SAVING;
                 case SAVING:
-                    if (writeFuture != null && !writeFuture.isDone())
+                    if (writeFuture != null && (!writeFuture.isDone() || !writeFuture.isSuccess()))
```

Comment:
when the future failed we treat this as success without checking for success; this can cause the tx to hang